From a8e2ae978dca12b309676ff87a3ccda90524ca0a Mon Sep 17 00:00:00 2001
From: Seungwoo Yang <ysw.yang@lge.com>
Date: Wed, 22 Jun 2016 12:35:12 +0900
Subject: [PATCH] Fix the wifi connection failure

:Release Notes:
Fix the wifi connection failure after connecting with the AP manually

:Detailed Notes:
After device connects to wifi router manually, if disconnecting this AP,
device cannot connect the AP again. When device disconnects the AP, connman
deletes the saved profile and IP address except the connected method
value in ipconfig structure. At this time, device tries to connect manually
without the IP address which causes connection failure. This change
will set the method value in ipconfig to CONNMAN_IPCONFIG_METHOD_DHCP,
when device disconnects the manual wifi network.

:Testing Performed:
Build successfully
Ethrenet/wifi connection successfully

:Issues Addressed:
[WOSLQEVENT-116424] Wifi connection failure
[WOSLQEVENT-116707] Wifi connection failure

Change-Id: Ib06f23b8c2cd5a4acbd8fb792156fb139539dd51
Reviewed-on: http://gpro.lgsvl.com/230653
Reviewed-by: Rajesh Kanna <rajesh.kanna@lge.com>
Tested-by: Rajesh Kanna <rajesh.kanna@lge.com>
---
 src/service.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/service.c b/src/service.c
index edbdd51d..327a8327 100644
--- a/src/service.c
+++ b/src/service.c
@@ -7440,6 +7440,11 @@ int __connman_service_disconnect(struct connman_service *service)
 		__connman_ipconfig_set_proxy_autoconfig(service->ipconfig_ipv6,
 							NULL);
 
+	if (service->type == CONNMAN_SERVICE_TYPE_WIFI &&
+			__connman_ipconfig_get_method(service->ipconfig_ipv4) == CONNMAN_IPCONFIG_METHOD_MANUAL) {
+		__connman_ipconfig_set_method(service->ipconfig_ipv4, CONNMAN_IPCONFIG_METHOD_DHCP);
+	}
+
 	__connman_ipconfig_address_remove(service->ipconfig_ipv4);
 	settings_changed(service, service->ipconfig_ipv4);
 
-- 
2.17.1

